### DISCUSSION TITLE
Error: "No code actions here" on clicking flymake overlays

### DISCUSSION THREAD

**Discussion by jdtsmith** [joaotavora/eglot#1071] [General]
A small but persistent problem I encounter with eglot is that mouse-1 is bound to `eglot-code-actions` on `flymake-note` overlays, even when there is no code action.  Mostly this is harmless, but clicking anywhere in the highlighted note text produces the error "[eglot] No code actions here".  This is correct: there are no code actions on this position.  But I don't see a need for this error, and it prevents placing cursor on the highlighted text with the mouse.   

This does not occur for, e.g., `flymake-error` (which doesn't appear to have the mouse-1 keymap set).

**Comment by joaotavora** [comment]
Thanks. I think I can replicate this. Related to #1053

**Comment by jdtsmith** [comment]
I had a typo in my config.  I am now using this:

```elisp
  (put 'eglot-note 'flymake-overlay-control nil)  ; was 'eglot-node
  (put 'eglot-warning 'flymake-overlay-control nil)
  (put 'eglot-error 'flymake-overlay-control nil)
```

which prevents the behavior for all three overlay flavors.  So that explains the non-uniformity.  A better solution might be to simply pass the click through if there is no code action at a given overlay, rather than signal an error.

**Reply by joaotavora** [reply]
> which prevents the behavior for all three overlay flavors. So that explains the non-uniformity. A better solution might be to simply pass the click through if there is no code action at a given overlay, rather than signal an error.

Conditional click-through is not easy to do, but it's possible.  Doesn't change the fact that one may just want to place point on it even with the code action.  Having bound `mouse-1` to it wasn't a good decision, it has to be changed.

**Reply by joaotavora** [reply]
This has been fixed in the latest Eglot. Please check if possible and report back here.
